ng-dev/misc/new-main-branch/remote-fork-update.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {debug, green, promptConfirm, promptInput, warn, yellow} from '../../utils/console';
+
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client';
+import {GithubRepo} from '../../utils/git/github';
+import {info} from 'console';
+
+/**
+ * Prompts the user whether remote forks for the current repository should be
+ * updated from `master` to `main`.
+ *
+ * An access token for performing the Github Admin operation is demanded through
+ * a prompt. This is opt-in as not every contributor may want to grant the tool a
+ * Github access token/or some contributors may want to make the changes themselves.
+ */
+export async function promptForRemoteForkUpdate() {
+  if (
+    !(await promptConfirm(
+      'Do you want to update your fork(s) on Github to also use `main`? (recommended)',
+    ))
+  ) {
+    return;
+  }
+
+  info('');
+  info(yellow('In order to be able to update your fork automatically, the script needs'));
+  info(yellow('authenticated access to your GitHub account. For this, you need to enter a'));
+  info(yellow('GitHub access token that is temporarily stored in memory until the script exits.'));
+  info('');
+
+  if (!(await promptConfirm('Do you want to proceed updating your forks automatically?'))) {
+    return;
+  }
+
+  const accessToken = await promptInput(
+    'Please enter a Github access token (`public_repo` scope is required)',
```

Comment:
Should we provide a url in the terminal?

`https://github.com/settings/tokens/new?scopes=public_repo&description=automated%20Angular%20repository%20primary%20branch%20renaming`

Reply:
oh yeah, that's a great idea. Didn't know this works